### PR TITLE
Hw 56015 change pagination

### PR DESCRIPTION
**Pull request by arao6**
Pagination changes for V4

### CODE REVIEW

**Review by Rrathinasabapath — APPROVED** [review]
Verified the changes and ran the tests successfully. Please raise the PR.

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletHateoasLink.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package com.hyperwallet.clientsdk.model;
+
+import java.util.List;
+import java.util.Map;
+
+public class HyperwalletHateoasLink{
```
Comment (peter-joseph) [inline src/main/java/com/hyperwallet/clientsdk/model/HyperwalletHateoasLink.java:6]: space before `{`

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletList.java [hunk 2]
```diff
@@ -5,26 +5,29 @@
...
+    private List<T> data;
+    private List<HyperwalletHateoasLink> links = new ArrayList<HyperwalletHateoasLink>();
 
-    private List<T> data = new ArrayList<T>();
+    public HyperwalletList() {
+    }
 
-    public int getCount() {
-        return count;
+    public boolean isHasNextPage() {
```
Comment (peter-joseph) [inline src/main/java/com/hyperwallet/clientsdk/model/HyperwalletList.java:17]: change to `hasNextPage()`

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletList.java [hunk 3]
```diff
@@ -5,26 +5,29 @@
...
 
-    public void setCount(int count) {
-        this.count = count;
+    public void setHasNextPage(boolean hasNextPage) {
+        this.hasNextPage = hasNextPage;
     }
 
-    public int getOffset() {
-        return offset;
+    public boolean isHasPreviousPage() {
```
Comment (peter-joseph) [inline src/main/java/com/hyperwallet/clientsdk/model/HyperwalletList.java:25]: change to `hasPreviousPage()`

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletHateoasLink.java (hunk 1 above)
Comment (peter-joseph) [inline src/main/java/com/hyperwallet/clientsdk/model/HyperwalletHateoasLink.java:6]: can we name this as `HyperwalletLink` and in Javadoc just declare that this represents `HATEOAS`

Comment (arao6) [reply src/main/java/com/hyperwallet/clientsdk/model/HyperwalletHateoasLink.java:6]: Updated the class name.

**Comment by arao6** [comment]
Review comments addressed and changes pushed into the branch.

**Pull request merged**